s3stream/src/main/java/com/automq/stream/s3/operator/DeleteObjectsAccumulator.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2024, AutoMQ CO.,LTD.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+package com.automq.stream.s3.operator;
+
+import com.automq.stream.s3.metrics.TimerUtil;
+import com.automq.stream.s3.metrics.stats.S3OperationStats;
+import com.automq.stream.utils.FutureUtil;
+import com.automq.stream.utils.Threads;
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class DeleteObjectsAccumulator implements Runnable {
+    static final Logger LOGGER = LoggerFactory.getLogger(DeleteObjectsAccumulator.class);
+    public static final int DEFAULT_DELETE_OBJECTS_MAX_BATCH_SIZE = 1000;
+    public static final int DEFAULT_DELETE_OBJECTS_MAX_CONCURRENT_REQUEST_NUMBER = 100;
+    private final ExecutorService batchDeleteCheckExecutor = Threads.newFixedThreadPoolWithMonitor(1,
```

Comment:
The accumulator could be optimized to without extra thread

Reply:
got it I'll try to optimize this.